tabby-terminal/src/api/baseTerminalTab.component.ts

```diff
@@ -788,6 +785,26 @@ export class BaseTerminalTabComponent<P extends BaseTerminalProfile> extends Bas
         })
     }
 
+    /**
+     * Method called when session is closed.
+     */
+    protected onSessionClosed (destroyOnSessionClose = false): void {
+        if (destroyOnSessionClose || this.doesTabShouldBeDestroyedOnSessionClosed()) {
+            this.destroy()
+        }
+    }
+
+    /**
+     * Return true if tab should be destroyed on session closed.
+     */
+    protected doesTabShouldBeDestroyedOnSessionClosed (): boolean {
+        const behavior = this.profile.behaviorOnSessionEnd
+        if (behavior === 'close' || behavior === 'auto' && this.isSessionExplicitlyTerminated()) {
```

Comment:
`return behavior === 'close' || behavior === 'auto' && this.isSessionExplicitlyTerminated()`